js/main.js

```diff
@@ -0,0 +1,45 @@
+//Happy Birthday Leader! 🎇💙
+
+const card_list = document.getElementById("card-list");
+const CLOUD_NAME = "dazcxdgiy";
+const CLOUDINARY_URL = `https://res.cloudinary.com/${CLOUD_NAME}/image/upload/`;
+
+//Custom Card component. Use it like this:
+//<tcg-card card-id="[CARD_ID]"></tcg-card>
+class Card extends HTMLElement {
+  constructor() {
+    super();
+    const image = document.createElement("img");
+    image.src = this.getImageURL();
+    image.classList.add("card-image");
+    this.appendChild(image);
+  }
+
+  //This returns an url of the form:
+  //`https://res.cloudinary.com/${CLOUD_NAME}/image/upload/cld-sample-${CARD_ID}`
+  //if "card-id" isn't supplied it returns a F L O W E R 🌸 sample image url.
+  getImageURL() {
+    return `${CLOUDINARY_URL}${
+      this.hasAttribute("card-id")
+        ? `cld-sample-${this.getAttribute("card-id")}`
+        : "sample.jpg"
+    }`;
+  }
+}
+
+//Renders some sample images from Cloudinary's sample folder using our Card component.
+function renderImages() {
+  for (let i = 2; i <= 5; i++) {
```

Comment:
Is there no way to get all the images (or id's) from a given folder via the cloudinary API? Otherwise I'm not sure if there's any effortless way we can keep this loop "synced" when new cards get added. We would have to manually update the code each time.

Reply:
There is a way to do it, but it requires setting tags for the images [(docs)](https://cloudinary.com/documentation/advanced_url_delivery_options#client_side_asset_lists). We would need to either do it manually or write a small script to do it. There's a way to get a list directly from folders or even from filenames [(docs)](https://cloudinary.com/documentation/search_api) , but it requires the API key and secret so it would need either showing them client side (a big no-no), or using Netlify Functions to act as a backend. We can also go the CSV route and just plop the CSV in the files and access it through javascript. This needs extended discussion.

Reply:
Good to know about the multiple options!